store/tikv/2pc.go

```diff
@@ -422,6 +414,24 @@ func (c *twoPhaseCommitter) commitSingleBatch(bo *Backoffer, batch batchKeys) er
 	return nil
 }
 
+func (c *twoPhaseCommitter) commitSingleBatch(bo *Backoffer, batch batchKeys) error {
+	// If we fail to receive response for the request that commits primary key, it will be undetermined whether this
+	// transaction has been successfully committed.
+	// Under this circumstance,  we can not declare the commit is complete (may lead to data lost), nor can we throw
+	// an error (may lead to the duplicated key error when upper level restarts the transaction). Currently the best
+	// solution is to populate this error and let upper layer drop the connection to the corresponding mysql client.
+	isPrimary := false
+	if bytes.Compare(batch.keys[0], c.primary()) == 0 {
+		isPrimary = true
+	}
+
+	if err := c.doCommitSingleBatch(bo, batch); err != nil && isPrimary {
+		// change the Cause of the error to be returned
+		return errors.Wrap(err, terror.ErrResultUndetermined)
+	}
+	return nil
```

Comment:
if it's canceled, will this return ErrResultUndetermined ?
how to distinguish between manual cancel and canceled caused by backoff timeout ?
@hhkbp2